src/components/connect.js

```diff
@@ -47,7 +37,6 @@ export default function connect(mapStateToProps, mapDispatchToProps, mergeProps,
 
   const finalMergeProps = mergeProps || defaultMergeProps
   const { pure = true, withRef = false } = options
-  const checkMergedEquals = pure && finalMergeProps !== defaultMergeProps
```

Comment:
I think the premise was that if we use `defaultMergeProps`, we *know* that it’s *not* going to be shallowly equal if either `stateProps` or `dispatchProps` changed. Therefore not even worth doing a `shallowEqual` check.